src/wolf.system/cmake/external.cmake

```diff
@@ -0,0 +1,240 @@
+### requires WOLF_EXTERN_DEPS_DIR and PREFIX_LIB_PATH
+
+include_guard()
+
+include(FetchContent)
+
+# A hack that lets FetchContent cooperate with find_package
+set(HANDLED_BY_EXTERNAL "") # a list of packages for find_package to ignore
+
+# Note: when you shadow a function, the previous definition
+# is still available by prefixing with an underscore
+
+# Idea from C++Now2017: Daniel Pfeifer "Effective CMake"
+# https://youtu.be/bsXLMQ6WgIk?t=3159
+# Note: there appears to be a typo in the presentation slides.
+# We need to check ${ARGV0}, not ${ARG0}.
+macro(find_package)
+    if(NOT "${ARGV0}" IN_LIST HANDLED_BY_EXTERNAL)
+        _find_package(${ARGV})
+    else()
+        message(DEBUG "${ARGV0} already fetched")
+    endif()
+endmacro()
+
+function(wolf_find_expat)
+    message(STATUS "wolf dependency libexpat")
+    FetchContent_Declare(
+        expat
+        GIT_REPOSITORY https://github.com/libexpat/libexpat.git
+        GIT_TAG        master
+        SOURCE_SUBDIR  expat
+    )
+    list(APPEND HANDLED_BY_EXTERNAL expat)
+    FetchContent_MakeAvailable(expat)
+endfunction()
+
+function(wolf_find_apr)
+    message(STATUS "wolf dependency apr")
+
+    set(APR_MINIMAL_BUILD ON CACHE BOOL "APR_MINIMAL_BUILD")
+
+    FetchContent_Declare(
+        apr
+        GIT_REPOSITORY https://github.com/apache/apr.git
+        GIT_TAG        trunk
+    )
+
+    list(APPEND HANDLED_BY_EXTERNAL apr)
+    FetchContent_MakeAvailable(apr)
+
+    FetchContent_GetProperties(apr)
+    target_include_directories(apr-2 PUBLIC "${apr_BUILD_DIR}")
+    target_include_directories(apr-2 PUBLIC "${apr_SOURCE_DIR}/include")
+endfunction()
+
+function(wolf_find_libcurl)
+    message(STATUS "wolf dependency libcurl")
+    FetchContent_Declare(
+        libcurl
+        GIT_REPOSITORY https://github.com/curl/curl.git
+        GIT_TAG        master
+    )
+    list(APPEND HANDLED_BY_EXTERNAL libcurl)
+    FetchContent_MakeAvailable(libcurl)
```

Comment:
Including `libcurl` header files into the project are forgotten. Please include these header files in the project.